### PR TITLE
Mam 189 sniffer refactor

### PR DESCRIPTION
**Pull request by fgmamoru**

### CODE REVIEW

File: mamoru-sniffer-c/headers.h [hunk 1]
```diff
@@ -14,14 +14,103 @@
 extern "C" {
```
Comment (sokyrko) [inline mamoru-sniffer-c/headers.h:14]: Looks like unrelated changes in this file.

File: mamoru-sniffer/src/sniffer.rs [hunk 2]
```diff
@@ -167,17 +167,17 @@ impl SnifferBgTask {
         message_client: MessageClient,
         query_client: QueryClient,
         daemons: Arc<RwLock<Vec<Daemon>>>,
-        chain_type: ChainType,
+        chain_type: &Vec<ChainType>,
```
Comment (sokyrko) [inline mamoru-sniffer/src/sniffer.rs:170]: In this code you pass a reference and then copy the underlying data. Pass by value instead: `Vec<ChainType>`.

File: mamoru-sniffer/src/sniffer.rs [hunk 3]
```diff
@@ -194,7 +194,7 @@ impl SnifferBgTask {
     /// Stops when incidents channel `report_rx` is dropped.
     pub(crate) async fn run(mut self) {
         debug!(
-            chain_type = self.chain_type.as_str_name(),
+            chain_type = self.chain_type[0].as_str_name(),
```
Comment (sokyrko) [inline mamoru-sniffer/src/sniffer.rs:197]: `chain_type`'s len may be 0 here. It's better to show all of them.

File: mamoru-sniffer/src/sniffer.rs [hunk 4]
```diff
@@ -245,7 +245,7 @@ impl SnifferBgTask {
     async fn update_daemons(&self) -> SnifferResult<()> {
         let daemon_response: Vec<DaemonQueryResponseDto> = self
             .query_client
-            .list_daemons(self.chain_type)
+            .list_daemons(self.chain_type[0]) // TODO: support multiple chain types, not sure what to do here.
```
Comment (sokyrko) [inline mamoru-sniffer/src/sniffer.rs:248]: A single sniffer supports a single chain type, as it reads a single blockchain network. I think `self.chain_type` should not be `Vec`.

File: mamoru-sniffer/src/validation_chain/message_client.rs [hunk 5]
```diff
@@ -104,16 +104,21 @@ impl MessageClient {
         })
     }
 
-    pub async fn register_sniffer(&self, chain: ChainType) -> ClientResult<()> {
+    pub async fn register_sniffer(&self, chains: Vec<ChainType>) -> ClientResult<()> {
```
Comment (sokyrko) [inline mamoru-sniffer/src/validation_chain/message_client.rs:107]: A single sniffer supports a single chain type, as it reads a single blockchain network.

**Review by sokyrko — CHANGES_REQUESTED** [review]

File: mamoru-sniffer-c/headers.h (hunk 1 above)
Comment (fgmamoru) [reply mamoru-sniffer-c/headers.h:14]: This seems to be related with Cargo fmt and clippy

File: mamoru-core/src/daemon/mod.rs [hunk 6]
```diff
@@ -95,76 +73,8 @@ impl Daemon {
...
-    fn daemon(active_since: i64, inactive_since: i64) -> Daemon {
-        Daemon::new(
-            "test".to_string(),
-            active_since,
-            inactive_since,
-            Box::new(TestExecutor),
-        )
+    pub fn is_active(&self) -> bool {
+        // @TODO: should this depend of staking?
+        true
```
Comment (sokyrko) [inline mamoru-core/src/daemon/mod.rs:78]: Remove this method, it's not relevant anymore.

File: mamoru-sniffer/Cargo.toml [hunk 7]
```diff
@@ -26,6 +26,7 @@ thiserror = "1.0"
 tokio = { workspace = true }
 tonic = { version = "0.8", features = ["tls-roots"] }
 tracing = "0.1"
+uuid = "1.3.0"
```
Comment (sokyrko) [inline mamoru-sniffer/Cargo.toml:29]: `uuid` should be in `[dev-dependencies]` section as it is used for tests only

File: mamoru-sniffer/src/validation_chain/message_client.rs [hunk 8]
```diff
@@ -186,10 +194,10 @@ impl MessageClient {
 
     pub async fn register_daemon(
         &self,
+        daemon_metadata_id: String,
         chain: ChainType,
-        content: impl Into<String>,
-        activate_since: chrono::DateTime<Utc>,
-        inactivate_since: chrono::DateTime<Utc>,
+        parameters: Vec<DaemonParameter>,
+        relay: DaemonRelay,
```
Comment (sokyrko) [inline mamoru-sniffer/src/validation_chain/message_client.rs:200]: I believe DaemonRelay is optional parameter

File: mamoru-sniffer/src/validation_chain/message_client.rs [hunk 9]
```diff
@@ -106,14 +107,15 @@ impl MessageClient {
...
+        }];
 
         self.sign_and_broadcast_txs(vec![MsgRegisterSniffer {
             creator: sniffer.clone(),
             sniffer: Some(SnifferRegisterCommandRequestDto {
                 sniffer,
-                chain: Some(Chain {
-                    chain_type: chain.into(),
-                }),
+                chains: chain_vec,
```
Comment (sokyrko) [inline mamoru-sniffer/src/validation_chain/message_client.rs:118]: `chains: vec![Chain { chain_type: chain.into() }]`

File: mamoru-sniffer/src/validation_chain/proto.rs [hunk 10]
```diff
@@ -24,21 +23,14 @@ impl TryFrom<DaemonQueryResponseDto> for Daemon {
...
-            .map_err(DaemonParseError::DateTime)?
-            .timestamp();
-
-        let rule = Self::new_sql(
+        let daemon = Self::new_sql(
             value.daemon_id,
-            activate_since,
-            inactivate_since,
-            &value.content,
+            &value.daemon_metadata.unwrap().content.unwrap().query[0]
```
Comment (sokyrko) [inline mamoru-sniffer/src/validation_chain/proto.rs:28]: Do we have a follow up task to fix this?

File: mamoru-sniffer/tests/validation_chain_tests/sniffer.rs [hunk 11]
```diff
@@ -55,17 +63,17 @@ async fn get_incidents(daemon_id: &str) -> Vec<IncidentQueryResponseDto> {
         .expect("List incidents error")
 }
 
-async fn register_active_daemon(chain: ChainType, query: impl Into<String>) -> String {
+async fn register_active_daemon(
```
Comment (sokyrko) [inline mamoru-sniffer/tests/validation_chain_tests/sniffer.rs:66]: It should be called just `register_daemon`, as they all active now

File: Readme.md [hunk 12]
```diff
@@ -1,12 +1,12 @@
...
 
-### Init submodules
+## Init submodules
 
 ```shell
 make init
 ```
 
-### Install development dependencies
+## Install development dependencies
```
Comment (sokyrko) [inline Readme.md:9]: Please don't change the layout.

File: mamoru-core/src/blockchain_data/mod.rs [hunk 13]
```diff
@@ -102,10 +102,6 @@ impl BlockchainDataCtx {
         &self.tx_hash
     }
 
-    pub(crate) fn time(&self) -> NaiveDateTime {
-        self.now
```
Comment (sokyrko) [inline mamoru-core/src/blockchain_data/mod.rs:106]: Is `self.now` still required?

**Review by sokyrko — COMMENTED** [review]
Thanks for the code. I've left a few more comments

File: mamoru-sniffer/src/validation_chain/message_client.rs (hunk 8 above)
Comment (fgmamoru) [reply mamoru-sniffer/src/validation_chain/message_client.rs:200]: daemonRelay can be set as type NONE, but seems to be mandatory

File: mamoru-sniffer/src/validation_chain/proto.rs (hunk 10 above)
Comment (fgmamoru) [reply mamoru-sniffer/src/validation_chain/proto.rs:28]: yes

File: mamoru-core/src/blockchain_data/mod.rs (hunk 13 above)
Comment (fgmamoru) [reply mamoru-core/src/blockchain_data/mod.rs:106]: removed

File: mamoru-sniffer/src/validation_chain/message_client.rs (hunk 9 above)
Comment (fgmamoru) [reply mamoru-sniffer/src/validation_chain/message_client.rs:118]: done in previous line

**Review by sokyrko — APPROVED** [review]

**Pull request merged**